pitop/robotics/drive_controller.py

```diff
@@ -66,48 +76,107 @@ def __calculate_motor_rpms(self, linear_speed, angular_speed, turn_radius):
+        """
         self.forward(-speed_factor, hold)
 
-    def left(self, speed_factor, turn_radius):
+    @is_initialized
+    def left(self, speed_factor, turn_radius=0):
+        """Make the robot move to the left, using a circular trajectory.
+
+        :param float speed_factor:
+            Factor relative to the maximum motor speed, used to set the velocity, in the range -1.0 to 1.0.
+            Using negative values will cause the robot to turn right.
+        :param float turn_radius:
+            Radius used by the robot to perform the movement. Using `turn_radius=0` will cause the robot to rotate in place.
+        """
+
         self.__robot_move(self._linear_speed_x_hold, self._max_robot_angular_speed * speed_factor, turn_radius)
 
-    def right(self, speed_factor, turn_radius):
+    @is_initialized
+    def right(self, speed_factor, turn_radius=0):
+        """Make the robot move to the right, using a circular trajectory.
+
+        :param float speed_factor:
+            Factor relative to the maximum motor speed, used to set the velocity, in the range -1.0 to 1.0.
+            Using negative values will cause the robot to turn left.
+        :param float turn_radius:
+            Radius used by the robot to perform the movement. Using `turn_radius=0` will cause the robot to rotate in place.
+        """
+
         self.left(-speed_factor, -turn_radius)
 
     def target_lock_drive_angle(self, angle):
         angular_speed = self.__target_lock_pid_controller(angle)
         self.__robot_move(self._linear_speed_x_hold, angular_speed)
 
-    def rotate(self, angle, angular_speed):
+    @is_initialized
+    def rotate(self, angle, time_to_take):
+        """Rotate the robot in place by a given angle and stop.
+
+        :param float angle: Angle of the turn.
+        :param float time_to_take: Expected duration of the rotation, in seconds.
+        """
+        assert time_to_take > 0.0
+        angle_radians = radians(angle)
+        angular_speed = angle_radians / time_to_take
+
         angular_speed = angular_speed * angle / abs(angle)
         rpm_left, rpm_right = self.__calculate_motor_rpms(0, angular_speed, turn_radius=0)
         rotations = abs(angle) * pi * self._wheel_separation / (360 * self._wheel_circumference)
-        self._left_motor.set_target_rpm(target_rpm=rpm_left,
-                                        total_rotations=rotations*rpm_left/abs(rpm_left))
-        self._right_motor.set_target_rpm(target_rpm=rpm_right,
-                                         total_rotations=rotations*rpm_right/abs(rpm_right))
+        self.left_motor.set_target_rpm(target_rpm=rpm_left,
+                                       total_rotations=rotations*rpm_left/abs(rpm_left))
+        self.right_motor.set_target_rpm(target_rpm=rpm_right,
+                                        total_rotations=rotations*rpm_right/abs(rpm_right))
+        sleep(time_to_take)
 
     def stop(self):
+        """Completely stops the robot."""
```

Comment:
Think of function docstrings as finishing the sentence "This function, when called, will..."

* Rotate the robot in place by a given angle and stop ✅
* Completely stops the robot ❌